### PR TITLE
Add data storage/retrieval capability and test cases for IliadDataController.

### PR DESCRIPTION
**Pull request by DersuG**
- Test cases are run from `test.py`, which calls `test()` on various other files. This is so relative imports work correctly.
- Data in `IliadDataController` is stored in lists of timestamp-value pairs.
- Packet data is split up. For example, `GPS_COORDINATES` packets are split into 2 lists, one for latitude and one for longitude.
- Other parts of the application can access these lists directly (ie. for graphing).

### CODE REVIEW

**Comment by DersuG** [comment]
Instead of running `test.py`, we will use `pytest` instead. Test cases have been rearranged/updated accordingly.

File: data_controllers/iliad_data_controller.py [hunk 1]
```diff
@@ -11,6 +10,76 @@ def __init__(self, identifier: str) -> None:
 
         self.data_buffer = bytearray()
         self.checksum_calculator = crc.CrcCalculator(crc.Crc16.CCITT)
+
+        self.arm_status_1_data: list[tuple[float, bool]] = []
+        self.arm_status_2_data: list[tuple[float, bool]] = []
+        self.arm_status_3_data: list[tuple[float, bool]] = []
```
Comment (PeterHall16) [inline data_controllers/iliad_data_controller.py:16]: What is arm status 3? Camera?

File: data_controllers/iliad_data_controller.py [hunk 2]
```diff
@@ -11,6 +10,76 @@ def __init__(self, identifier: str) -> None:
...
+        self.altitude_1_data: list[tuple[float, float]] = []
+        self.altitude_2_data: list[tuple[float, float]] = []
+        self.acceleration_x_data: list[tuple[float, float]] = []
+        self.acceleration_y_data: list[tuple[float, float]] = []
+        self.acceleration_z_data: list[tuple[float, float]] = []
```
Comment (PeterHall16) [inline data_controllers/iliad_data_controller.py:21]: There might be another set of acceleration fields, not sure yet

**Review by PeterHall16 — COMMENTED** [review]
Left some minor comments

File: data_controllers/iliad_data_controller.py (hunk 1 above)
Comment (DersuG) [reply data_controllers/iliad_data_controller.py:16]: Not sure, I just followed the specification for that packet type. Are we going to have arming logic for the camera board?

File: data_controllers/iliad_data_controller.py (hunk 2 above)
Comment (DersuG) [reply data_controllers/iliad_data_controller.py:21]: We can add them later

File: data_controllers/iliad_data_controller.py (hunk 1 above)
Comment (PeterHall16) [reply data_controllers/iliad_data_controller.py:16]: There is no camera board, it's part of the FC. Treating it like an arming status is probably the easiest way to deal with it though

**Review by PeterHall16 — COMMENTED** [review]
Looks good

**Review by PeterHall16 — APPROVED** [review]
CLosing PR

**Pull request merged**